content/2022/videookulografija.md

```diff
@@ -1,4 +1,561 @@
+
+
+
+
+
+Zbog toga linija se posmatra u parametarskom obliku, a formula glasi:
+
+
+$x \cdot \cos{\theta} + y \cdot \sin{\theta} = \rho$,
+
+gde je:   
+- $\theta$ - ugao koji zaklapa normala na pravu sa x osom Slika 13
+- $\rho$ - rastojanje od koordinantnog početka do prave
+- x, y - parametri
+
+
+Sada, tačka neće predstavljati pravu već sinusoidu Slika 13, a presečne tačke se određuju na isti način kao što je ranije navedeno.
+
+
+![Tačka u sinusoidnom obliku - slika 13](/images/2022/videookulografija/hafovatransformacija2.png)
+
+(slika 13 - Tačka u sinusoidnom obliku)
+
+
+
+
+U narednom koraku se definiše parametarski prostor.
+Za svaki tačku iz slike, koja je detektovana kao ivica se generiše sinusoida koja se pamti i traže se preseci. 
+Parametarski prostor se određuje pomoći ro i teta. 
+Definišu se minimalne i maksimalne veličine, tj. opsezi za svaki od dva data parametra. Potom se taj parametarski prostor deli na više ćelija u skladu sa potrebnom preciznošću. 
+U sledećem koraku se za svak tačku sa originalne slike definiše prava, nakon čega se za svaku vrednost parametra $\theta$ određuje parametar $\rho$. 
+Ako važi da parametar $\theta_a$ generiše $\rho_b$, onda se ćelija inkrementira. 
+Za sve tačke je postupak isti. 
+Cilj je izdvojiti ćelije koje imaju najveće vrednosti, zbog toga što linije koje prolaze kroz te ćelije imaju najviše tačaka. 
+
+
+Hafova transformacija za detekciju krugova, kao što je rečeno, identična je transformaciji linije. Kod krugova istih poluprečnika, tačke iz dekartovog se u parametarski sistem slikaju kao kružnice. 
+Dakle, potrebna su samo dva parametra a i b, što sužava pretragu na dve dimenzije. 
+Krug koji se predstavlja jednačinom kružnice
+
+$(x - a)^2 + (y - b)^2 = r^2$ , 
+
+parametarski se predstavlja formulama: 
+
+$x = a + r \cdot \cos{\theta}$,
+
+$y = b + r \cdot \sin{\theta}$. 
+
+
+Kod krugova različitih poluprečnika razlika je u tome što oni zahtevaju tri dimenzije, tj. u parametarski prostor se prenose kao kupe. 
+Ostatak postupka je isti kao i za LHT.
+
+
+![3D transformacija kruga - slika 14](/images/2022/videookulografija/hafovatransformacija2.png)
+
+(slika 14 - 3D transformacija kruga)
+
+
+#### **3.2.2. Obrada podataka**
+
+#### 3.2.2.1. Mean average filter
+
+
+Dok konstantno gledamo u jednu tačku dolazi do šuma kao što je treptaj, nenamerno pomeranje oka ili kamere, promena jačine osvetljenja i slično.
+Ovakve šumove neophodno je otkloniti i u te svrhe koristimo mean avrage filter.
+Prolazići kroz signal izračunava se aritmetička vrednost prethodnih N merenja koordinata zenice.
+Ovako izračunate vrednosti se postavlju kao nove koordinate zenice. 
+Na taj način se dobija filtrirano merenje.
+
+Na slikama ispod se mogu videti grafovi filtriranih signala, x i y ose u odnosu na vreme.
+
+
+![Koordinate x-ose u odnosu na t - slika 15](/images/2022/videookulografija/filtriranx.jpg)(slika 13)
```

Comment:
ose su premale, baš se ne vide. nema legende da se zna šta je šta. Ako je moguće, ispravite to